core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1134,6 +1139,7 @@ public void failShard(String reason, @Nullable Exception e) {
 
     private Engine.Searcher acquireSearcher(String source, Engine.SearcherScope scope) {
         readAllowed();
+        lastSearcherAccess.lazySet(threadPool.relativeTimeInMillis());
```

Comment:
I spent some cycles reading this and convincing my self that we can do this regardless of the searcher scope - i.e., that if we end up here with an internal searcher it will be OK (it is!). I would personally think it will be simpler to follow if we mark access in `awaitPendingRefresh`. That would mean all flow /state management is around the engine and you don't have to go through engine code to see how things work. This is of course subjective, so just a suggestion.

Reply:
we have to do this in here otherwise we can't make a decision if a shard is idle or not. sorry your comment doesn't make sense with respect to the engine. the engine is not involved here